share/p2p/discovery/discovery.go

```diff
@@ -79,7 +78,7 @@ func NewDiscovery(
 		connector:      newBackoffConnector(h, defaultBackoffFactory),
 		onUpdatedPeers: func(peer.ID, bool) {},
 		params:         params,
-		triggerDisc:    make(chan struct{}),
+		triggerDisc:    make(chan struct{}, 1), // buffer to memoize discovery request
```

Comment:
so do we think the problem here is that signal is sent when it's not ready to read? Is this the fix here or is the findPeersTimeout the fix? From what i understand from our discussion, it seems like the latter

Reply:
This buffer is an opportunistic fix. Honestly, there is no understanding of why it would help, but there is some evidence from @walldiss that it helps. We need to reconfirm this I think

Reply:
Based on my understanding of logs attached to original issue, is that at some moment discovery lost peers and wasn't triggered, but it should have been. It wasn't stuck at that point either, because there were no specific log lines for that. There was another period of time when it was stuck and there were periodic logs for that. 
So PR is addressing those 2 plus another one, that Advertise could error and retry time was low.

Reply:
Just realised that I didn't answered your Q 😆 . I have no evidence why discovery wasn't triggered, but my only guess is that it could be because the signal was lost due to no reader on other side of the channel at the time of signal.

Reply:
So I won't block on this because it's not _really_ a big deal to buffer the channel but it does suck bc I don't fully understand why this exact change is necessary yet. The best thing would've been to test the two solutions separately and then together so we could see which fix precisely is the actual fix but 🤷🏻

Reply:
@walldiss, it would be nice if you could do that, specifically test the node without the buffer

Reply:
FindPeers was resulting in periodic logs, we had this case covered and now it should be never the case with your fix. There was another issue that has not resulted in such logs, so it wasn't stuck in FindPeers, but rather wasn't starting.